packages/graphql-hooks/src/useClientRequest.js

```diff
@@ -158,6 +158,15 @@ function useClientRequest(query, initialOpts = {}) {
         if (revisedOpts.useCache) {
           actionResult.useCache = true
           actionResult.cacheKey = revisedCacheKey
+
+          if (client.ssrMode) {
+            const cacheValue = {
+              data: revisedOpts.updateData
+                ? revisedOpts.updateData(state.data, actionResult.data)
```

Comment:
SSR still doesn't work when options.updateData is passed because `state.data` is undefined here.

[![Edit with-graphql-hooks](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/with-graphql-hooks-nfgxb?fontsize=14)

Reply:
@Raincal If you're up for it we'd welcome a PR - ideally with a breaking test commit that highlights the issue, then the fix :)